### DISCUSSION TITLE
$persist with Alpine.data

### DISCUSSION THREAD

**Discussion by stefanfrede** [alpinejs/alpine#3661] [7. Bugs (archived)]
:wave:,

I use $persist with Alpine.data and my method to close a banner is not triggered anymore.

Here I init Alpine.data:
https://github.com/stefanfrede/11st-starter-kit/blob/main/src/main.js#L10

And here is my component:
https://github.com/stefanfrede/11st-starter-kit/blob/main/src/_includes/components/site-banner.webc

When I click the button my dismiss method is not triggered.

Everything else seems to work fine.

Has something changed because it definitely worked at some point.

Thanks for a push in the right direction to fix this!

**Comment by SimoTod** [comment]
It works correctly for me: https://codepen.io/SimoTod/pen/ExOErwL

**Comment by stefanfrede** [comment]
Hmmm, this is weird. Then it seems I have a side effect which kills my button. Thank you!

**Comment by stefanfrede** [comment]
Is it possible that I have problems because I use Alpine.js in an Eleventy WebC template?

I tried now some basic stuff with Alpine.js in my WebC template and it is not working.

**Comment by stefanfrede** [comment]
I found it. It seems WebC is removing stuff when generating the HTML.
